docs/sources/mimir/release-notes/v2.10.md

```diff
@@ -0,0 +1,106 @@
+- **Reduce wasted effort spent computing results that won't be used** by making queriers cancel the requests sent to a zone upon receiving first error from that zone.
+- **Added the ability to clean up the object storaage by the compactor** when no blocks left in the bucket index by removing the bucket index, markers and debug files. This cleanup can be enabled by using `-compactor.no-blocks-file-cleanup-enabled` option.
+- **Added new debug HTTP endpoints** `/ingester/tenants` and `/ingester/tsdb/{tenant}` to the ingester that provide debug information about tenants and their TSDBs.
+
+Additionally, the following previously experimental features are now considered stable:
+
+- **Support for a ruler storage cache**. This cache should reduce the number of "list objects" API calls issued to the object storage when there are 2+ ruler replicas running in a Mimir cluster. The cache can be configured by setting the `-ruler-storage.cache.*` CLI flags or their respective YAML config options.
+- **Query sharding cardinality estimation.** You can enable it through the advanced CLI configuration flag `-query-frontend.query-sharding-target-series-per-shard`, allows query sharding to take into account cardinality of similar requests executed previously when computing the maximum number of shards to use. If you want to try it out, we recommend starting with a value of `2500`.
+- **Query expression size limit**. You can limit the size in bytes of the queries allowed to be processed through the CLI configuration flag `-query-frontend.max-query-expression-size-bytes`.
+- **Peer discovery / tenant sharding for overrides exporters**. You can enable it through the CLI configuration flag `-overrides-exporter.ring.enabled`.
+- **Overrides exporter enabled metrics selection**. You can select which metrics the overrides exporter should export through the CLI configuration flag `-overrides-exporter.enabled-metrics`.
+- **Per-tenant results cache TTL**.
+
+## Experimental features
+
+Grafana Mimir 2.10 includes new features that are considered as experimental and disabled by default. Please use them with caution and report any issues you may encounter:
+
+- **Support for ingesting exponential histograms in OpenTelemetry format**. The exponential histograms that are over the native histogram scale limit of 8 can be downscaled to allow their ingestion.
+- **Store-gateway index-header loading improvements**, which include the ability to persist the sparse index-header to disk instead of reconstructing it on every restart (`-blocks-storage.bucket-store.index-header-sparse-persistence-enabled`) and an option to limit the number of concurrent index-header loads when lazy-loading (`-blocks-storage.bucket-store.index-header-lazy-loading-concurrency`).
```

Comment:
You can also mention persisting the list of block IDs that were lazy-loaded while running. Same can be eagerly loaded upon startup to prevent starting up with no loaded blocks. The config option is `-blocks-storage.bucket-store.index-header.eager-loading-startup-enabled`